gdal/frmts/tiledb/tiledbdataset.cpp

```diff
@@ -0,0 +1,1492 @@
+
+        if( psOldTree == nullptr )
+            psOldTree = CPLCreateXMLNode( nullptr, CXT_Element, "PAMDataset" );
+
+        for( psSubTree = psOldTree->psChild;
+             psSubTree != nullptr;
+             psSubTree = psSubTree->psNext )
+        {
+            if( psSubTree->eType != CXT_Element
+                || !EQUAL(psSubTree->pszValue, "Subdataset") )
+                continue;
+
+            if( !EQUAL(CPLGetXMLValue( psSubTree, "name", "" ),
+                       psPam->osSubdatasetName) )
+                continue;
+
+            break;
+        }
+
+        if( psSubTree == nullptr )
+        {
+            psSubTree = CPLCreateXMLNode( psOldTree, CXT_Element,
+                                          "Subdataset" );
+            CPLCreateXMLNode(
+                CPLCreateXMLNode( psSubTree, CXT_Attribute, "name" ),
+                CXT_Text, psPam->osSubdatasetName );
+        }
+
+        CPLXMLNode *psOldPamDataset = CPLGetXMLNode( psSubTree, "PAMDataset");
+        if( psOldPamDataset != nullptr )
+        {
+            CPLRemoveXMLChild( psSubTree, psOldPamDataset );
+            CPLDestroyXMLNode( psOldPamDataset );
+        }
+
+        CPLAddXMLChild( psSubTree, psTree );
+        psTree = psOldTree;
+    }
+
+/* -------------------------------------------------------------------- */
+/*      Try saving the auxiliary metadata.                               */
+/* -------------------------------------------------------------------- */
+
+    CPLPushErrorHandler( CPLQuietErrorHandler );
+
+    int bSaved = 0;
+    vfs.touch( psPam->pszPamFilename );
+    tiledb::VFS::filebuf fbuf( vfs );
+    fbuf.open( psPam->pszPamFilename, std::ios::out );
+    std::ostream os(&fbuf);
+
+    if (os.good())
+    {
+        char* pszTree = CPLSerializeXMLTree( psTree );
+        os.write( pszTree, strlen(pszTree));
+        bSaved = 1;
+    }
+
+    fbuf.close();
+    
+    CPLPopErrorHandler();
+
+/* -------------------------------------------------------------------- */
+/*      If it fails, check if we have a proxy directory for auxiliary    */
+/*      metadata to be stored in, and try to save there.                */
+/* -------------------------------------------------------------------- */
+    CPLErr eErr = CE_None;
+
+    if( bSaved )
+        eErr = CE_None;
+    else
+    {
+        const char *pszBasename = GetDescription();
+
+        if( psPam->osPhysicalFilename.length() > 0 )
+            pszBasename = psPam->osPhysicalFilename;
+
+        const char *pszNewPam = nullptr;
+        if( PamGetProxy(pszBasename) == nullptr
+            && ((pszNewPam = PamAllocateProxy(pszBasename)) != nullptr))
+        {
+            CPLErrorReset();
+            CPLFree( psPam->pszPamFilename );
+            psPam->pszPamFilename = CPLStrdup(pszNewPam);
+            eErr = TrySaveXML();
+        }
+        /* No way we can save into a /vsicurl resource */
+        else if( !STARTS_WITH(psPam->pszPamFilename, "/vsicurl") )
```

Comment:
It would be unusual with this driver to use any of `/vsi` however it is technically possible with setting a proxy directory for the auxiliary metadata and `/vsicurl` is blacklisted in the main code for this write.